The new version breaks Win64 compilation with MinGW64

The newly ported string hashing implementation breaks the build on 64-bit Windows using MinGW64:

```
C:\msys64\home\agentzh\work\openresty-1.19.3.1rc0\build\LuaJIT-2.1-20201002\src/x64/src/lj_str_hash_x64.h:158: undefined reference to `srandom'
C:\msys64\home\agentzh\work\openresty-1.19.3.1rc0\build\LuaJIT-2.1-20201002\src/x64/src/lj_str_hash_x64.h:167: undefined reference to `random'
C:\msys64\home\agentzh\work\openresty-1.19.3.1rc0\build\LuaJIT-2.1-20201002\src/x64/src/lj_str_hash_x64.h:168: undefined reference to `random'
C:\msys64\home\agentzh\work\openresty-1.19.3.1rc0\build\LuaJIT-2.1-20201002\src/x64/src/lj_str_hash_x64.h:179: undefined reference to `random'
C:\msys64\home\agentzh\work\openresty-1.19.3.1rc0\build\LuaJIT-2.1-20201002\src/x64/src/lj_str_hash_x64.h:179: undefined reference to `random'
collect2.exe: error: ld returned 1 exit status
```

@siddhesh Will you have a quick look at it? Thanks!

--- a/src/x64/src/lj_str_hash_x64.h
+++ b/src/x64/src/lj_str_hash_x64.h
@@ -21,6 +21,11 @@
 #undef LJ_AINLINE
 #define LJ_AINLINE
 
+#ifdef __MINGW32__
+#define random()  ((long) rand())
+#define srandom(seed)  srand(seed)
+#endif
+
 static const uint64_t* cast_uint64p(const char* str)
 {
   return (const uint64_t*)(void*)str;

